src/values/AbstractValue.js

```diff
@@ -74,6 +75,7 @@ export default class AbstractValue extends Value {
   values: ValuesDomain;
   mightBeEmpty: boolean;
   args: Array<Value>;
+  reactHint: void | ReactHint;
```

Comment:
I am decidedly less than thrilled with this kind of addition, but a better alternative does not readily present itself.

Reply:
I thought about storing a `WeakMap` referencing AbstractValue's to their reactHints. I opted for this approach, but if anyone else can think of a better approach, I'd be happy to change.

Reply:
@NTillmann @sebmarkbage @cblappert what is your take on this PR?

Reply:
A comment on the meaning of the `reactHint` would nice. I am a bit lost.